var/spack/repos/builtin/packages/datatransferkit/package.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Datatransferkit(CMakePackage):
+    """DataTransferKit is an open-source software library of
+    parallel solution transfer services for multiphysics simulations"""
+
+    homepage = "datatransferkit.readthedoc.io"
+    url      = "https://github.com/ORNL-CEES/DataTransferKit/archive/3.1-rc1.tar.gz"
+    git      = "https://github.com/ORNL-CEES/DataTransferKit.git"
+
+    version('master', branch='master', submodules=True)
+
+    variant('openmp', default=False, description='enable OpenMP backend')
+    variant('serial', default=True, description='enable Serial backend (default)')
+
+    depends_on('cmake', type='build')
+    depends_on('trilinos@develop:+intrepid2+shards', when='+serial')
+    depends_on('trilinos@develop:+intrepid2+shards+openmp', when='+openmp')
```

Comment:
These have to be updated to have `trilinos~dtk`, to make sure it does not install Trilinos with a conflicting DTK.

Reply:
I'm also not sure why this only depends on `trilinos@develop`. I would assume that at least it works with `master` too, if you don't want to rely on any Trilinos version.

Reply:
~dtk is added to prevent potential conflict;

For the reason I pick trilinos@develop, I followed the package file for trilinos:
in trilinos/package.py:
line 218: branch='master'
line 221: when='+dtk @develop'

Reply:
I don't think this is necessary here. And I think it is likely that Trilinos should be changed too. In fact, I think all DTK versions past 2 should be removed from the Trilinos package.

Reply:
I think that should be done only once this new package is part of the xSDK test suite.